### PR TITLE
Update express-middleware.js

### LINKED ISSUE #61: Error running with express and nextjs - TypeError [ERR_INVALID_ARG_TYPE]: The "chunk" argument must be of type string or an instance of Buffer. Received an instance of Promise
I'm using express with nextjs, but pulled the express app out.  Code look llike so

```js
// server.js 
const next = require('next');
const { initializeExpressWithNext } = require('./src/server/app');
const dev = process.env.NODE_ENV !== 'production';
const nextApplication = next({ dev });
const handle = nextApplication.getRequestHandler();

nextApplication.prepare().then(() => {
  const app = initializeExpressWithNext(handle);

  app.listen(3000, (err) => {
    if (err) throw err;
    console.log('Ready on http://localhost:3000');
  });
});


//app.js
const express = require('express');
const cors = require('cors');
const apiMetrics = require('prometheus-api-metrics');

const questionnaireRoutes = require('./routes/questionnaire');
const recaptchaRoutes = require('./routes/recaptcha');

/**
 * @function initializeExpressWithNext
 * @description to run this with nextjs, we need to pass in a callbackhandler
 * for all routes that are not defined by express, we do this by wrapping express
 * in a function to initialize the callback and runtime
 * @param {function} callbackHandler - nextjs server callbackhandler
 * @return {*} express application
 */
function initializeExpressWithNext(callbackHandler) {
  const app = express();
  app.use(apiMetrics());

  app.get('/healthcheck', cors(), (_, res) => {
    res.status(200).send();
  });

  app.use('/', questionnaireRoutes);
  app.use('/', recaptchaRoutes);

  app.get('*', (req, res) => {
    return callbackHandler(req, res);
  });

  return app;
}

module.exports = { initializeExpressWithNext };
```

The app starts fine, but when i navigate to /metrics or metrics.json, I get the following error

```
TypeError [ERR_INVALID_ARG_TYPE]: The "chunk" argument must be of type string or an instance of Buffer. Received an instance of Promise
    at ServerResponse.end (_http_outgoing.js:784:13)
    at ExpressMiddleware.middleware (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/prometheus-api-metrics/src/express-middleware.js:99:24)
    at Layer.handle [as handle_request] (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/index.js:317:13)
    at /Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/index.js:275:10)
    at expressInit (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/middleware/init.js:40:5)
    at Layer.handle [as handle_request] (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/index.js:317:13)
    at /Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/index.js:275:10)
    at query (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/middleware/query.js:45:5)
    at Layer.handle [as handle_request] (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/nathanhall/bwell/medstar-squeeze-page/node_modules/express/lib/router/index.js:317:13)
    ```
    
    I'll try to see if I can figure it out and make. PR but I'm on a tight timeline

### PR DESCRIPTION
**Pull request by dominathan**
prom-client returns Promise.  Need to handle promise in express middleware.
Resolves #61

### CODE REVIEW

File: test/unit-test/metric-middleware-test.js [hunk 1]
```diff
@@ -378,14 +378,18 @@ describe('metrics-middleware', () => {
...
-            sinon.assert.calledWith(set, 'Content-Type', Prometheus.register.contentType);
-            sinon.assert.calledOnce(set);
+
+            setTimeout(() => {
+                sinon.assert.calledOnce(end);
+                // eslint-disable-next-line no-control-regex
+                const endFormalized = end.getCall(0).args[0].replace(/ ([0-9]*[.])?[0-9]+[\x0a]/g, ' #num\n');
+                // eslint-disable-next-line no-control-regex
+                let apiFormalized = Prometheus.register.metrics()
+                apiFormalized = apiFormalized.replace(/ ([0-9]*[.])?[0-9]+[\x0a]/g, ' #num\n');
```
Comment (jbarnaby-medallia) [inline test/unit-test/metric-middleware-test.js:388]: `apiFormalized` is a promise, so it doesn't have a `replace` method.. you have to make the test case async I would think

Comment (yorch) [reply test/unit-test/metric-middleware-test.js:388]: `apiFormalized` is a promise, so it doesn't have a replace method.. you have to make the test case async I would think

**Comment by yorch** [comment]
Found this issue today when tried to upgrade `prom-client`. You would also need to update koaMiddleware and a bunch of tests.

**Comment by dominathan** [comment]
Yea, I'm using my own version of prom client now with express.  If someone ones to pick this up feel free, I'm going to close this.

**Comment by yorch** [comment]
That's the spirit!

https://github.com/PayU/prometheus-api-metrics/pull/65/files

**Pull request closed without merging**